Analysis

mohammadbashiri opened the pull request:
Added display analysis with visualization to the analysis script. As well as using the method in the cmd api to display the result right after running a display experiment

mohammadbashiri committed:
updating the notebook

mohammadbashiri committed:
working on the notebook and integrating the methods into analysis script

mohammadbashiri committed:
working on the notebook and integrating the methods into analysis script

mohammadbashiri committed:
using the display method in the cmd_api

nickdelgrosso commented on vrlatency/analysis.py:
```diff
@@ -104,59 +104,99 @@ def get_display_dataframe(filename):
     session = df.Session.values[0]
-    latencies = df.groupby('Session').apply(get_display_latencies, thresh=.75).unstack()
+    thresh = .75
+    latencies = df.groupby('Session').apply(get_display_latencies, thresh=thresh).unstack()
```
Do we need to group by sessions anymore?  This should be a single-session analysis

nickdelgrosso commented on vrlatency/analysis.py:
```diff
@@ -104,59 +104,99 @@ def get_display_dataframe(filename):
+    return dd
+
+
 def display_brightness_figure(filename, ax1=None, ax2=None):
```
This function is trying to do too much.  File reading, data transformations, and dataframe handling should be removed.

nickdelgrosso commented on vrlatency/analysis.py:
```diff
@@ -104,59 +104,99 @@ def get_display_dataframe(filename):
+
 def display_brightness_figure(filename, ax1=None, ax2=None):
 
+    want_to_draw = not bool(ax1 and ax2)
```
want_to_draw is a strange variable name for a figure-drawing function.  It points to the function trying to do too much.

nickdelgrosso reviewed (changes requested):
The analysis functions are quite overloaded, and should be made more specific.  Some examples are available in this review.

mohammadbashiri committed:
refactoring the analysis script

mohammadbashiri committed:
refactoring the analysis script

mohammadbashiri committed:
refactoring the analysis script

mohammadbashiri committed:
refactoring the analysis script

mohammadbashiri committed:
no o of the plotting functions take Pandas DataFrame as an input now

mohammadbashiri committed:
working on the report notebook

mohammadbashiri committed:
no o of the plotting functions take Pandas DataFrame as an input now

mohammadbashiri committed:
changes to notebook, as well as changing samplesize for arduino mega

mohammadbashiri committed:
resolved merge conflict

mohammadbashiri committed:
resolved merge conflict

mohammadbashiri committed:
changes from origin master

mohammadbashiri committed:
updated the display experiment

mohammadbashiri committed:
Merge branch 'analysis' of https://github.com/mohammadbashiri/vrlatency into analysis

mohammadbashiri committed:
recording display experiment

mohammadbashiri committed:
Arduino is crazy fast now!

mohammadbashiri committed:
trials now get clustered based on the SSE

mohammadbashiri committed:
trials now get clustered based on the SSE

mohammadbashiri committed:
testing different display signal acquisition with the faster arduino

mohammadbashiri committed:
analysis pipeline runs now without error.
on finding the index of the trough of residuals, sometimes we are faced with a situation we the slope of the brightness transtion is different. and if the test signal crosses the ref signal, then shifting in any direction will increase SSE and as a result we will not have a trough. in that case we should just take the index of the minimum residual, which also means no offset.

With regards to the length of the signals for computing the residuals, the length of the reference signal is now forced to be equal to the test signal
